### PR TITLE
Refactor creating block messages for list action

### PR DESCRIPTION
**Pull request by parberge**
Create Slack class methods for creating block messages in a generic way to make it easier to use blocks.
Remove the old `create_block_message` function

#132

### CODE REVIEW

File: chalicelib/action.py [hunk 1]
```diff
@@ -350,3 +347,24 @@ def _valid_reason(self, reason: str) -> bool:
             return True
 
         return False
+
+    def _create_list_message(self, data) -> None:
+        """
+        Create the slack block message layout for list action
+        """
+        data = json.loads(data)
```
Comment (parberge) [inline chalicelib/action.py]: Just a thought; Do you think it's better that this method expects a native data type instead of json?

Comment (kamigerami) [reply chalicelib/action.py]: In the end I think the blocks are json.. so it is fine with json actually makes it easier.

File: chalicelib/action.py [hunk 2]
```diff
@@ -350,3 +347,26 @@ def _valid_reason(self, reason: str) -> bool:
...
+        start_date = data[0].get("event_date")
+        end_date = data[-1].get("event_date")
+
+        self.slack.add_section_block(
+            text=f"Reported time for period *{start_date}:{end_date}*",
+        )
+        self.slack.add_divider_block()
+
+        for event in data:
+            start_date = event.get("event_date")
```
Comment (kamigerami) [inline chalicelib/action.py]: dont we have this already at the top? `start_date = data[0].get("event_date")`

Comment (parberge) [reply chalicelib/action.py]: This is per event, but yeah the name of the variable can be better ☺️

**Pull request merged**